src/client/Microsoft.Identity.Client/ManagedIdentity/V2/MtlsBindingStore.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Identity.Client.Core;
+
+namespace Microsoft.Identity.Client.ManagedIdentity.V2
+{
+    /// <summary>
+    /// Centralized helper for installing, locating and pruning IMDSv2 client mTLS binding certs
+    /// from the CurrentUser\My store.
+    /// </summary>
+    internal static class MtlsBindingStore
+    {
+        // Minimum remaining lifetime required to reuse a binding.
+        internal static readonly TimeSpan MinFreshRemaining = TimeSpan.FromMinutes(5);
+
+        /// <summary>
+        /// Installs the certificate in CurrentUser\My, removes any exact-duplicate thumbprints,
+        /// and best-effort prunes older certs for the same subject. Returns the subject DN used
+        /// as lookup key.
+        /// </summary>
+        internal static string InstallAndGetSubject(X509Certificate2 cert, ILoggerAdapter logger = null)
```

Comment:
Why do we have an `MTLS Cert Store` and this Binding Store that both deal with certs and stores? Seems like a lot of dup code.